packages/ember-glimmer/tests/integration/helpers/get-test.js

```diff
@@ -0,0 +1,337 @@
+    }));
+
+    this.assertText('[green] [green]');
+  }
+
+  ['@test should be able to get an object value with nested static key']() {
+    this.render(`[{{get colors "apple.gala"}}] [{{if true (get colors "apple.gala")}}]`, {
+      colors: { apple: { gala: 'red and yellow' }, banana: 'yellow' }
+    });
+
+    this.assertText('[red and yellow] [red and yellow]');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('[red and yellow] [red and yellow]');
+
+    this.runTask(() => set(this.context, 'colors', {
+      apple: { gala: 'yellow and red striped' }, banana: 'purple'
+    }));
+
+    this.assertText('[yellow and red striped] [yellow and red striped]');
+
+    this.runTask(() => set(this.context, 'colors.apple.gala', 'yellow-redish'));
+
+    this.assertText('[yellow-redish] [yellow-redish]');
+  }
+
+  ['@test should be able to get an object value with a bound/dynamic key']() {
+    this.render(`[{{get colors key}}] [{{if true (get colors key)}}]`, {
+      colors: { apple: 'red', banana: 'yellow' },
+      key: 'apple'
+    });
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'key', 'banana'));
+    this.assertText('[yellow] [yellow]');
+
+    this.runTask(() => set(this.context, 'colors', {
+      apple: 'green',
+      banana: 'purple'
+    }));
+    this.assertText('[purple] [purple]');
+
+    this.runTask(() => set(this.context, 'key', 'apple'));
+    this.assertText('[green] [green]');
+
+    this.runTask(() => set(this.context, 'colors.apple', 'red'));
+    this.assertText('[red] [red]');
+  }
+
+  ['@test should be able to get an object value with nested dynamic key']() {
+    this.render(`[{{get colors key}}] [{{if true (get colors key)}}]`, {
+      colors: {
+        apple: {
+          gala: 'red and yellow',
+          mcintosh: 'red'
+        },
+        banana: 'yellow'
+      },
+      key: 'apple.gala'
+    });
+    this.assertText('[red and yellow] [red and yellow]');
+
+    this.runTask(() => set(this.context, 'key', 'apple.mcintosh'));
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'key', 'banana'));
+    this.assertText('[yellow] [yellow]');
+  }
+
+  ['@test should be able to get an object value with subexpression returning nested key']() {
+    this.render(`[{{get colors (concat 'apple' '.' 'gala')}}] [{{if true (get colors (concat 'apple' '.' 'gala'))}}]`, {
+      colors: {
+        apple: {
+          gala: 'red and yellow',
+          mcintosh: 'red'
+        },
+        banana: 'yellow'
+      },
+      key: 'apple.gala'
+    });
+    this.assertText('[red and yellow] [red and yellow]');
+
+    this.runTask(() => set(this.context, 'colors', { apple: { gala: 'yellow and red striped' }, banana: 'purple' }));
+    this.assertText('[yellow and red striped] [yellow and red striped]');
+
+    this.runTask(() => set(this.context, 'colors.apple.gala', 'yellow-redish'));
+    this.assertText('[yellow-redish] [yellow-redish]');
```

Comment:
Either add a "reset with replacement" step after this or change this to test that